### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by iotcloudarchitect**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [ x] I have read the **CONTRIBUTING** document
- [ x] My code follows the code style of this project
- [ x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

Added documentation to Piral Inspector.

### Remarks

[Optionally place any follow-up comments, remarks, observations, or notes here for future reference]

### CODE REVIEW

File: src/pages/docs/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "test": "echo \"Error: run tests from root\" && exit 1"
   },
   "dependencies": {
+    "flexsearch": "^0.6.32",
```
Comment (FlorianRappl) [inline src/pages/docs/package.json:36]: Should not be there.

File: docs/commands/debug-piral.md [hunk 2]
```diff
@@ -143,15 +143,15 @@ Automatically installs missing Node.js packages.
 Also includes the node modules for target transpilation.
 
 - Type: `boolean`
-- Default: `false`
+- Default: `true`
```
Comment (FlorianRappl) [inline docs/commands/debug-piral.md:146]: Should be `false`. Don't know why this was changed here; same is true for the others.

Comment (iotcloudarchitect) [reply docs/commands/debug-piral.md]: Fixed. Docgen has been performed before bootstrapping updates from upstream. Sorry.

File: docs/tools/piral-inspector.md [hunk 3]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+- Collection of emitted events
+- Collection of global states
+
+As a developer of your piral instance, or your pilets, you are used to working with a web browser. And this is precisely the best place to get assisted by the Piral Inspector during the development or testing phase.
+
+We provide the Piral Inspector as an extension for the following browsers:
+- Google Chrome (>= v50)
+- Firefox (>= v50)
+- Opera (>= v50)
+
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:23]: Nitpick: No consecutive blank lines (just drop one).

File: docs/tools/piral-inspector.md [hunk 4]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+
+## Installation of add-on / extension
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:25]: Title case: "Installation of Add-On / Extension"

File: docs/tools/piral-inspector.md [hunk 5]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+Type the following URL into the browser address line followed by hitting enter:
+```
+about:addons
+```
+
+On the following page, you can ```search for "Piral Inspector"```, and you will see the add-on as the first result:
+
+![FirefoxAddOnSearchResultPiralInspector](../diagrams/piralinspector_firefox_add-on_searchresult.png)
+
+Click on ```"Piral Inspector"```.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:38]: If its inline then make single backtick. If its a code block then make three. Here its just a single one.

(Actually, I think here its just a string that is not in code - so just use `"Piral Inspector"`)

File: docs/tools/piral-inspector.md [hunk 6]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+- Firefox (>= v50)
+- Opera (>= v50)
+
+
+## Installation of add-on / extension
+
+### Firefox
+
+Type the following URL into the browser address line followed by hitting enter:
+```
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:30]: plaintext

File: docs/tools/piral-inspector.md [hunk 7]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+- Google Chrome (>= v50)
+- Firefox (>= v50)
+- Opera (>= v50)
+
+
+## Installation of add-on / extension
+
+### Firefox
+
+Type the following URL into the browser address line followed by hitting enter:
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:29]: Newline after (always have a blank line between blocks).

File: docs/tools/piral-inspector.md [hunk 8]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+about:addons
+```
+
+On the following page, you can ```search for "Piral Inspector"```, and you will see the add-on as the first result:
+
+![FirefoxAddOnSearchResultPiralInspector](../diagrams/piralinspector_firefox_add-on_searchresult.png)
+
+Click on ```"Piral Inspector"```.
+
+On the following details page, click on ```"Add to Firefox"```:
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:40]: As above - no need for the backticks. Same applies to all the phrases below.

File: docs/tools/piral-inspector.md [hunk 9]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+- Information about Piral instance
+- Available Pilets
+- Add Pilets
+- Registered Routes
+- Events
+- State Container
+
+All of these sections will documented in the following chapters:
+
+### Piral instance
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:125]: Piral Instance

File: docs/tools/piral-inspector.md [hunk 10]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+#### State Container Logging
+Activates/Deactivates the logging of changes in the state container to the Developer Tools console.
+
+#### Load available Pilets
+Activates/Deactivates the loading of the available Pilets.
+> This option is only available when the user is debugging a Pilet.
+
+#### Full refresh on change
+Refreshes the current browser page in case there was a change in the debugged Pilet.
+> This option is only available when the user is debugging a Pilet.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:143]: Instead of a quote, consider using a box. I would use the warning: https://docs.piral.io/reference/documentation/test

```md
::: warning: Only for Pilets
This feature has an impact only when the emulator is used, i.e., when `pilet debug` is running.
```

File: docs/tools/piral-inspector.md [hunk 11]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+All of these sections will documented in the following chapters:
+
+### Piral instance
+In the very left top, the extension shows the name and the version of the currently running Piral instance.
+Besides that, in the very right top, a click on the icon allows the user to adapt the extension's settings to its needs.
+
+![PiralInstanceToolPiralInspector](../diagrams/piralinspector_tool_piralinstance.png)
+
+In the debug setting, the user can activate/deactivate the following options:
+![DebugSettingsToolPiralInspector](../diagrams/piralinspector_tool_debugsettings.png)
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:132]: Newline in between.

File: docs/tools/piral-inspector.md [hunk 12]
```diff
@@ -4,4 +4,215 @@ title: Piral Inspector
...
+Besides that, the user can emit a new event:
+
+![EmitEventToolPiralInspector](../diagrams/piralinspector_tool_emitevent.png)
+
+### State Container
+The final section shows the currently available global state (read-only) for the following topics:
+
+![StateContainerToolPiralInspector](../diagrams/piralinspector_tool_statecontainer.png)
+
+#### app
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:203]: I don't think the documentation should go into this. After all, everything that is displayed here may be changed (the state container can be fully defined / modified by the Piral instance owner) + other things may appear here (e.g., by some plugin or custom substate).

Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Removed.

File: docs/tools/piral-inspector.md (hunk 11 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md:131]: Done.

File: docs/tools/piral-inspector.md (hunk 10 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done. Thanks for the hint.

File: docs/tools/piral-inspector.md (hunk 9 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done.

File: docs/tools/piral-inspector.md (hunk 7 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md:28]: Done.

File: docs/tools/piral-inspector.md (hunk 8 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done.

File: docs/tools/piral-inspector.md (hunk 6 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done.

File: docs/tools/piral-inspector.md (hunk 5 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done.

File: docs/tools/piral-inspector.md (hunk 4 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Done.

File: docs/tools/piral-inspector.md (hunk 3 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md:23]: Done.

File: docs/tools/piral-inspector.md [hunk 13]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+- Firefox (>= v50)
+- Opera (>= v50)
+
+## Installation of Add-On / Extension
+
+::: summary: Firefox
+
+Type the following URL into the browser address line followed by hitting enter:
+
+about:addons
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:30]: Missing backticks (why did you remove them - my suggestion was to just add `plaintext` as the language)

File: docs/tools/piral-inspector.md [hunk 14]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+
+On the following details page, click on "Add to Firefox":
+ 
+![FirefoxAddOnAddToFirefoxPiralInspector](../diagrams/piralinspector_firefox_add-on_add.png)
+
+Now, a security message will appear, please click on "Add" to accept the required security settings:
+
+![FirefoxAddOnSecurityMessagePiralInspector](../diagrams/piralinspector_firefox_add-on_security.png)
+
+Piral Inspector is now available, and you can open it via the Web Developer Tools (```Tools > Web Developer > Toggle Tools```).
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:46]: Analog den anderen changes -> bitte backticks entfernen.

File: docs/tools/piral-inspector.md [hunk 15]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+
+In case you are currently running a Piral Instance in debug mode, you will see the available Pilets and registered routes.
+
+![FirefoxAddOnToolPiralInspector](../diagrams/piralinspector_firefox_add-on_tool.png)
+:::
+
+::: summary: Google Chrome
+
+Type the following URL into the browser address line followed by hitting enter:
+```
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:58]: plaintext (and newline above)

File: docs/tools/piral-inspector.md [hunk 16]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+
+In case you are currently running a Piral Instance in debug mode, you will see the available Pilets and registered routes.
+
+![GoogleChromeExtensionToolPiralInspector](../diagrams/piralinspector_chrome_add-on_tool.png)
+:::
+
+::: summary: Opera
+
+Type the following URL into the browser address line followed by hitting enter:
+```
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:84]: plaintext (and newline above)

File: docs/tools/piral-inspector.md [hunk 17]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+### Add Pilets
+There are two options to load additional Pilets into the currently running Piral Instance:
+
+#### Add a feed address
+Enter a feed address into the text field and click on the "Add" button.
+In case you chose a feed you have access to, the corresponding Pilets will be loaded and shown in the tile section and the list of available Pilets. Per default, the new Pilets will be active.
+
+#### Bundle local Pilet and upload it
+First, you need to "Browse" to the Pilet package file and then "Upload" it.
+In case you uploaded a proper Pilet, it will be loaded and shown in the tile section and the list of available Pilets. Per default, the new Pilet will be active, but will be given a random name.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:178]: New paragraph - newline in between.

File: docs/tools/piral-inspector.md [hunk 18]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+By clicking on the switch button on the left, the user can activate and deactivate each Pilet.
+
+With a click on the "X" on the right, the user can remove each Pilet individually.
+
+### Add Pilets
+There are two options to load additional Pilets into the currently running Piral Instance:
+
+#### Add a feed address
+Enter a feed address into the text field and click on the "Add" button.
+In case you chose a feed you have access to, the corresponding Pilets will be loaded and shown in the tile section and the list of available Pilets. Per default, the new Pilets will be active.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:174]: Also here there are many newlines missing. Just orient on how the other md files have been written.

File: docs/tools/piral-inspector.md [hunk 19]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+
+The origin will also be displayed when hovering over a tile.
+
+
+### Available Pilets
+The Piral Inspector lists all the currently available Pilets. Like illustrated in the table below.
+
+![AvailablePiletsToolPiralInspector](../diagrams/piralinspector_tool_availablePilets.png)
+
+By clicking on the switch button on the left, the user can activate and deactivate each Pilet.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:165]: Pilet should be written like "pilet" (lowercase), while Piral is written like "Piral" (uppercase).

File: docs/tools/piral-inspector.md [hunk 20]
```diff
@@ -4,4 +4,203 @@ title: Piral Inspector
...
+Activates/Deactivates the logging of changes in the state container to the Developer Tools console.
+
+#### Load available Pilets
+Activates/Deactivates the loading of the available Pilets.
+::: warning: Only for Pilets
+This feature has an impact only when the emulator is used, i.e., when `pilet debug` is running.
+:::
+
+#### Full refresh on change
+Refreshes the current browser page in case there was a change in the debugged Pilet.
```
Comment (FlorianRappl) [inline docs/tools/piral-inspector.md:143]: lowercase pilet. (everywhere)

File: docs/tools/piral-inspector.md (hunk 19 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md:165]: Pilet is name like Piral is.
What is the difference?
Lothar is curious about the difference.

Comment (FlorianRappl) [reply docs/tools/piral-inspector.md:165]: We discussed that a long time ago. Actually, even without an explanation "consistency" should be key (look at *all* the documentation).

Piral is a proper name (like "Bono") while pilet is a concept / normal name (like "microservice").

File: docs/tools/piral-inspector.md (hunk 13 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md:31]: Got it.

File: docs/tools/piral-inspector.md (hunk 14 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Got it.

File: docs/tools/piral-inspector.md (hunk 15 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Okay.

File: docs/tools/piral-inspector.md (hunk 16 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Okay.

File: docs/tools/piral-inspector.md (hunk 20 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Okay.

File: docs/tools/piral-inspector.md (hunk 18 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Okay.

File: docs/tools/piral-inspector.md (hunk 17 above)
Comment (iotcloudarchitect) [reply docs/tools/piral-inspector.md]: Okay.
